### ISSUE TITLE
Check for nonzero values in check_guest_interruptibility_state_sti_mov_ss

### ISSUE THREAD

**Issue by connojd** [Bareflank/hypervisor#227]
We need to check 

``` cpp
if (sti && mov_ss)
    throw ...
```

rather than

``` cpp
if (sti == 1 && mov_ss == 1)
    throw ...
```

since for that check we just care about whether the individual bits are set or not.

**Comment by rianquinn** [comment]
Wouldn't it then be:

```
if (!sti || !mov_ss)
    throw ...
```

**Comment by connojd** [comment]
The check fails when they're both set

**Comment by rianquinn** [comment]
I see it now. Ok... sounds good.

**Comment by connojd** [comment]
Fixed in #343
